From 53c01b783b8e3223ff589a794742e02647a5de98 Mon Sep 17 00:00:00 2001
From: yangyanchao <yangyanchao6@huawei.com>
Date: Wed, 28 Oct 2020 17:02:42 +0800
Subject: [PATCH] huawei-support-riscv

---
diff --git a/webrtc/rtc_base/system/arch.h b/webrtc/rtc_base/system/arch.h
index ed216e6..8a35eec 100644
--- a/webrtc/rtc_base/system/arch.h
+++ b/webrtc/rtc_base/system/arch.h
@@ -44,6 +44,13 @@
 #define WEBRTC_ARCH_32_BITS
 #endif
 #define WEBRTC_ARCH_LITTLE_ENDIAN
+#elif defined(__riscv)
+#define WEBRTC_ARCH_LITTLE_ENDIAN
+#if __riscv_xlen == 64
+#define WEBRTC_ARCH_64_BITS
+#else
+#define WEBRTC_ARCH_32_BTIS
+#endif
 #elif defined(__pnacl__)
 #define WEBRTC_ARCH_32_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN

-- 
